rclpy/src/rclpy/_rclpy.c

```diff
@@ -4763,6 +4975,10 @@ static PyMethodDef rclpy_methods[] = {
     "rclpy_take_response", rclpy_take_response, METH_VARARGS,
     "rclpy_take_response."
   },
+  {
+    "rclpy_take_event", rclpy_take_event, METH_VARARGS,
+    "Get the pending data for a ready QoS Event."
+  },
```

Comment:
This file looks huge, time to break into one file per python function or something? (at least partially split this)

Reply:
It is huge - but i'm kind of in favor of big code files for feature sets - single compilation unit usually builds faster than code with lots of headers and you always know where to find what you're looking for. Regardless - I'd be hesitant to break it out as part of this feature - that feels like its own dedicated change.